app/src/main/java/nerd/tuxmobil/fahrplan/congress/utils/SessionPropertiesFormatter.kt

```diff
@@ -0,0 +1,45 @@
+package nerd.tuxmobil.fahrplan.congress.utils
+
+import nerd.tuxmobil.fahrplan.congress.models.Session
+
+class SessionPropertiesFormatter {
+
+    fun getFormattedSpeakers(session: Session) =
+        with(session.speakers) {
+            when {
+                isEmpty() -> ""
+                else -> joinToString(", ")
+            }
+        }
+
+    fun getFormattedTrackLanguageText(session: Session) =
+        with(StringBuilder()) {
+            append(session.track)
+            if (session.track.isNotEmpty() && session.language.isNotEmpty()) {
+                append(" ")
+            }
+            if (session.language.isNotEmpty()) {
+                append("[")
+                append(getLanguageText(session))
+                append("]")
+            }
+            toString()
+        }
+
+    fun getLanguageText(session: Session) =
+        with(session.language) {
```

Comment:
The use of `with()` feels weird. I'd go with this:

```kotlin
    fun getLanguageText(session: Session): String {
        if (session.language.isEmpty()) {
            return ""
        }

        return session.language
                .replace("-formal", "")
                .replace("German", "de")
                .replace("german", "de")
                .replace("Deutsch", "de")
                .replace("deutsch", "de")
                .replace("English", "en")
                .replace("english", "en")
                .replace("Englisch", "en")
                .replace("englisch", "en")
    }
```

Reply:
A matter of style I would say. But I am fine with your version.